src/NHibernate/Impl/FutureBatch.cs

```diff
@@ -63,20 +53,42 @@ public IEnumerable<TResult> GetEnumerator<TResult>()
 			return new DelayedEnumerator<TResult>(() => GetCurrentResult<TResult>(currentIndex));
 		}
 
-		private void GetResults()
+		public IFutureValueAsync<TResult> GetFutureValueAsync<TResult>(CancellationToken cancellationToken)
+		{
+			int currentIndex = index;
+			return new FutureValueAsync<TResult>(() => GetCurrentResultAsync<TResult>(currentIndex, cancellationToken));
```

Comment:
Strange pattern here. This method and the next one are not async, but already takes a cancellation token, which will be used inside the async delegate actually performing the loading when the future value will be accessed.

Why not simply using a cancellation token supplied where we await?

This is even more strange with the `IAsyncEnumerable` case, where its implementation ends up having a cancellation token while performing an await depending on another cancellation token hidden in the delegate it uses.